modules/maps.js

```diff
@@ -1060,20 +1064,28 @@ function autoMap(hdStats, vmStatus) {
     if (game.options.menu.repeatVoids.enabled != 0) toggleSetting('repeatVoids');
 
     //Prestige
-    if ((getPageSetting('ForcePresZ') >= 0) && (game.global.world >= getPageSetting('ForcePresZ'))) {
-        const prestigeList = ['Supershield', 'Dagadder', 'Megamace', 'Polierarm', 'Axeidic', 'Greatersword', 'Harmbalest', 'Bootboost', 'Hellishmet', 'Pantastic', 'Smoldershoulder', 'Bestplate', 'GambesOP'];
-        needPrestige = prestigeList.some(prestige => game.mapUnlocks[prestige].last <= (game.global.world - 5));
-        //needPrestige = (offlineProgress.countMapItems(game.global.world) !== 0); TODO - Test this!
-    } else
-        needPrestige = prestige != "Off" && game.mapUnlocks[prestige] && game.mapUnlocks[prestige].last <= (game.global.world - 5) && game.global.challengeActive != "Frugal";
+    //needPrestige = (offlineProgress.countMapItems(game.global.world) !== 0); TODO - Test this!
+    const forcePrestigeZ = getPageSetting('ForcePresZ');
+    let prestigeMapLevel = z;
+    if (prestige !== 'Off') {
+        const prestigeUnlock = game.mapUnlocks[prestige];
+        prestigeMapLevel = prestigeUnlock.last + 5;
+        needPrestige = game.upgrades[prestige].allowed && prestigeUnlock && prestigeMapLevel <= z;
+    } else if ((forcePrestigeZ >= 0) && (game.global.world >= forcePrestigeZ)) {
+        for (const p of prestigeList) {
+            const prestigeUnlock = game.mapUnlocks[p];
+            if (!needPrestige && game.upgrades[p].allowed && prestigeUnlock && (prestigeUnlock.last + 5) <= z) {
+                prestigeMapLevel = prestigeUnlock.last + 5;
+                needPrestige = true;
+            }
+        }
```

Comment:
Here, if prestige is set to something, it'll ignore forcePrestigeZ, no?
forcePrestige is supposed to run a prestige map until it gets all the prestiges, and run it again for every new prestige that becomes available

Reply:
You're right, thank you! I'll fix it in the main branch